### DISCUSSION TITLE
Restrict faction that will respawn

### DISCUSSION THREAD

**Discussion by ThornbirdSitcom** [Vdauphin/HeartsAndMinds#1474] [Question and answer]
Hi!

I'm modifying the mission but I get a problem. I can't clearly understand how to generate certain faction AI. I mean, I need to load numbers of mods, but it is expected to generate only certain one faction. For example, I'm loading RHS, Project OPFOR and Blabla, but I'm expecting only spawn Takistani Armed forces.

I know it might be weird, but I need to reserve some possibilities for my friends to use various of equipments. I've tried first way in documents, but in vain. Once the mission begins to generate enemies, there is only errors instead of ai. Maybe I made some mistakes. As for the second way, I'm worried about units from RHS will also be generated.

Honestly I'm not a good guy in tech T_T but I really need help. Thank you!

**Comment by ThornbirdSitcom** [comment]
`//Save class name to global variable
btc_enemy_side = ["LOP_TKA"];
btc_type_units = ["LOP_TKA_Infantry_AA","LOP_TKA_Infantry_Corpsman","LOP_TKA_Infantry_Engineer","LOP_TKA_Infantry_GL","LOP_TKA_Infantry_AT","LOP_TKA_Infantry_MG","LOP_TKA_Infantry_Marksman","LOP_TKA_Infantry_Rifleman","LOP_TKA_Infantry_Rifleman_2","LOP_TKA_Infantry_Rifleman_3","LOP_TKA_Infantry_TL"];
btc_type_divers = _allclasse select 2;
btc_type_crewmen = ["LOP_TKA_Infantry_Crewman"];
btc_type_boats = _allclasse select 4;
btc_type_motorized = ["LOP_TKA_UAZ","LOP_TKA_UAZ_Open","LOP_TKA_Ural","LOP_TKA_Ural_open"];
btc_type_motorized_armed = ["LOP_TKA_UAZ_AGS","LOP_TKA_UAZ_DshKM","LOP_TKA_UAZ_SPG","LOP_TKA_BTR60"];
btc_type_mg = ["LOP_TKA_Static_DshKM","LOP_TKA_NSV_TriPod"];
btc_type_gl = ["LOP_TKA_Static_AGS17","LOP_TKA_Static_SPG9","LOP_TKA_Static_AT4"];`
Here is how I edit the sqf. Maybe I got something wrong? T_T

**Reply by Vdauphin** [reply]
Hello,
Project OPFOR is no more maintain, please use an up to date mod like 3cb factions at least.
Check error in your rpt and fix them one by one
> Type %localappdata%\Arma 3 into the file explorer address bar and upload the latest RPT file here using + on the left.

Cheers

**Reply by ThornbirdSitcom** [reply]
I found the root of the problems. It seems that the form I edited is wrong.
`btc_enemy_side = EAST;
btc_type_units = ["LOP_TKA_Infantry_AA","LOP_TKA_Infantry_Corpsman","LOP_TKA_Infantry_Engineer","LOP_TKA_Infantry_GL","LOP_TKA_Infantry_AT","LOP_TKA_Infantry_MG","LOP_TKA_Infantry_Marksman","LOP_TKA_Infantry_Rifleman","LOP_TKA_Infantry_Rifleman_2","LOP_TKA_Infantry_Rifleman_3","LOP_TKA_Infantry_TL"];
btc_type_divers = _allclasse select 2;
btc_type_crewmen = "LOP_TKA_Infantry_Rifleman";
btc_type_boats = _allclasse select 4;
btc_type_motorized = ["LOP_TKA_UAZ","LOP_TKA_UAZ_Open","LOP_TKA_Ural","LOP_TKA_Ural_open"];
btc_type_motorized_armed = ["LOP_TKA_UAZ_AGS","LOP_TKA_UAZ_DshKM","LOP_TKA_UAZ_SPG","LOP_TKA_BTR60"];
btc_type_mg = ["LOP_TKA_Static_DshKM","LOP_TKA_NSV_TriPod"];
btc_type_gl = ["LOP_TKA_Static_AGS17","LOP_TKA_Static_SPG9","LOP_TKA_Static_AT4"];
`
Now it works fine. I wonder if you can edit the 'change faction of ai' part of the document? I mean, it will be better if you can give some examples? You see, the types of variables here are different, probably leading to confusion and misunderstand.
As for the Project OPFOR, I have to use that. I know 3CB is better, however, I'm editing this for my group. You know, Arma is a multiplayer game, so I can't make all decisions. I'm trying to suggest. That's all I can do.
Anyway, thanks for you advice and help sincerely. Hopefully this mission will be better and better!

**Reply by Vdauphin** [reply]
You can at any time create a PR to improve the documentation but as I see, you found the correct way
